### PR TITLE
Refactory Enum and changed variables names

### PR DESCRIPTION
**Pull request by Noivo**
Refactor `Enum.reduce` because the function was doing too many things
 - Created new function `tickers_list` to put `Enum.reduce` logic

Some variables name weren't explicit
 - Changed the name for better read of the code

### CODE REVIEW

File: lib/demo/coinbase.ex [hunk 1]
```diff
@@ -26,29 +21,27 @@ defmodule Demo.Coinbase do
     |> aggregate_id(id)
   end
 
-  defp aggregate_id({:ok, response}, id) do
-    {:ok, Map.merge(response, %{"id" => id})}
+  defp tickers_list(products) do
+    Enum.reduce(products, [], fn product, list ->
+      fetch_products_tickers(product)
```
Comment (ZePedroResende) [inline lib/demo/coinbase.ex:26]: This is fetching a product ticker why call it produtcts tickers ?

File: lib/demo/coinbase.ex [hunk 2]
```diff
@@ -26,29 +21,27 @@ defmodule Demo.Coinbase do
...
+      |> parse_products_tickers(list)
+    end)
   end
 
-  defp aggregate_id(error, _) do
-    error
+  defp aggregate_id({:ok, ticker}, id) do
```
Comment (ZePedroResende) [inline lib/demo/coinbase.ex:31]: For what i understand you aren't agregating anything , you are adding an id right ?

File: lib/demo/coinbase.ex (hunk 1 above)
Comment (ZePedroResende) [reply lib/demo/coinbase.ex:26]: Why not use pattern match on the annonymous function and remove this fetch_product_ticker that is just a product(id) ? or remove the anonymous and put it in a normal function da receives a product and a list ?

File: lib/demo/coinbase.ex [hunk 3]
```diff
@@ -26,29 +21,27 @@ defmodule Demo.Coinbase do
...
-  defp parse_products_tickers({:ok, response}, acc) do
-    acc ++ [response]
-  end
+  defp aggregate_id(error, _), do: error
 
-  defp parse_products_tickers(_, acc) do
-    acc
-  end
+  defp parse_products_tickers({:ok, ticker}, list),
+    do: list ++ [ticker]
```
Comment (ZePedroResende) [inline lib/demo/coinbase.ex:38]: Replace this for with [ticker | list]

iex> list = [1, 2, 3]
iex> [0 | list]   # fast
[0, 1, 2, 3]
iex> list ++ [4]  # slow
[1, 2, 3, 4]

**Comment by ZePedroResende** [comment]
Please provide a more descritive name for the pr and add a why and how to the pr description

File: lib/demo/coinbase.ex (hunk 1 above)
Comment (Noivo) [reply lib/demo/coinbase.ex:26]: > This is fetching a product ticker why call it produtcts tickers ?

Right, the correct name should be **fetch_product_ticker**, I confused with parse function

File: lib/demo/coinbase.ex [hunk 4]
```diff
@@ -23,32 +18,26 @@ defmodule Demo.Coinbase do
...
+    |> add_id(id)
   end
 
-  defp aggregate_id({:ok, response}, id) do
-    {:ok, Map.merge(response, %{"id" => id})}
+  defp tickers_list(products) do
+    Enum.reduce(products, [], fn %{"id" => id}, list ->
+      product(id)
+      |> parse_products_tickers(list)
+    end)
```
Comment (ZePedroResende) [inline lib/demo/coinbase.ex:28]:
``` elixir
def products_ticker(products) do
	Enum.reduce(products, [], &product_ticker/2)
end

defp  product_ticker(%{"id" => id}, list) do
	with {:ok, product} <- product(id) do
		[product | list]	
	else
		list
	end
end
```

Couldn't this just be reduce to this ? Just so we don't use anonymous functions .

File: lib/demo/coinbase.ex [hunk 5]
```diff
@@ -23,32 +18,26 @@ defmodule Demo.Coinbase do
   def product(id) do
     get("products/" <> id <> "/ticker")
     |> parse_response()
-    |> aggregate_id(id)
+    |> add_id(id)
```
Comment (ZePedroResende) [inline lib/demo/coinbase.ex:21]: Should this really be here ? Couldn't this be done on the callers side ?

File: lib/demo/coinbase.ex [hunk 6]
```diff
@@ -7,12 +7,7 @@ defmodule Demo.Coinbase do
 
   def products do
     with {:ok, response} <- get("/products/") |> parse_response() do
-      result_list =
-        response
-        |> Enum.reduce([], fn product, acc ->
-          fetch_products_tickers(product)
-          |> parse_products_tickers(acc)
-        end)
+      result_list = tickers_list(response)
```
Comment (ZePedroResende) [inline lib/demo/coinbase.ex:10]: Should the products function return the tickers list ? 

This seems like a different call to the API that depends on the products call

**Comment by ZePedroResende** [comment]
@DavideSilva  if you have spare time could you take a look at the review , don't know if I'm being to opinionated

File: lib/demo/tracker.ex [hunk 7]
```diff
@@ -44,16 +44,17 @@ defmodule Demo.Tracker do
...
-          state        
+        error ->
+          IO.inspect(error)
+          state
       end
+
     head = List.first(state)
 
-    broadcast({:ok, head}, :history) 
+    broadcast({:ok, head}, :history)
```
Comment (ZePedroResende) [inline lib/demo/tracker.ex:57]: Should we call the broadcast event :history ? 
Wouldn't new_rates or something telling that its the new conversions ?

Comment (Noivo) [reply lib/demo/tracker.ex:57]: Sorry for not answer the others questions, I didn't notice on a reply already open on top so couldn't fine the button to reply.

new_rates looks good

Comment (Noivo) [reply lib/demo/tracker.ex]: I have already changed on the other pull request but new_rates seems better than last_conversions_rates

**Pull request closed without merging**